tox_conda/plugin.py

```diff
@@ -1,19 +1,187 @@
+        basepython = envconfig.basepython
+        if not basepython.startswith("python"):
+            msg = "{0} will not be created using conda, only Python basepython is supported by tox-conda.".format(
+                env
+            )
+            warn(RuntimeWarning(msg))
+            envconfig.conda = False
+            continue
+        envconfig.conda = True
+
+        # Split Conda dependencies from pip dependencies and create environment.yaml.
+        split_deps(envconfig)
+        create_env_yml(envconfig)
+
+
+@hookimpl
+def tox_testenv_create(venv, action):
+    """Perform creation action for this venv."""
+
+    if not venv.envconfig.conda:
+        return None
+
+    env_location = str(venv.path)
+    yaml = venv.envconfig.conda_envyaml
+    redirect = venv.session.config.option.verbose_level < 2
+    result = []
+
+    # Remove old environment if recreating.
+    if venv.envconfig.recreate:
+        if os.path.exists(env_location):
+            shutil.rmtree(env_location)
+
+    # Create new environment.
+    args = ["conda", "env", "update", "-p", env_location, "--file", yaml]
+    env = venv._getenv(testcommand=False)
+    result.append(action.popen(args, env=env, redirect=redirect))
+
+    return True if not result else result
+
+
+@hookimpl
+def tox_testenv_install_deps(venv, action):
+    """Perform install dependencies action for this venv."""
+    # If created using conda, all pip dependencies are already installed at creation.
+    if venv.envconfig.conda:
+        return True
+
+
+def get_conda_python(envconfig):
+    return os.path.join(envconfig.envdir, "python")
+
+
+def is_exe(fpath):
+    """Check whether a filepath points to an executable."""
+    return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
+
+
+def which(cmd):
+    """Try to find command on path and return full path if found."""
+    fpath, fname = os.path.split(cmd)
+    if fpath:
+        if is_exe(cmd):
+            return cmd
+    else:
+        for path in os.environ["PATH"].split(os.pathsep):
+            exe_file = os.path.join(path, cmd)
+            if is_exe(exe_file):
+                return exe_file
+
+    return None
+
+
+def split_deps(envconfig):
+    """Split conda and pip dependencies.
+
+    Pip dependencies are left in envconfig.deps.
+    Conda dependencies are put in envconfig.conda_deps.
+    """
+    python_version = envconfig.basepython[6:]
+    conda_deps = ["python={0}".format(python_version)]
+    pip_deps = []
+    for dep in envconfig.deps:
+        dep_str = str(dep)
+        if dep_str.startswith(("-r", "-e", "-c", ":")):
+            pip_deps.append(dep)
+        elif dep_str.startswith("--pip"):
+            dep.name = dep_str[5:].strip()
+            pip_deps.append(dep)
+        else:
+            conda_deps.append(dep)
+
+    envconfig.deps = pip_deps
+    envconfig.conda_deps = conda_deps
+
+
+def create_env_yml(envconfig):
```

Comment:
While creating a file is clever, I think it may be somewhat over-complicated. In my opinion, it would be easier to simply provide a list of `conda` dependencies to a `conda install` command in `tox_testenv_install_deps`. I think that the `pip` dependencies should be handled by calling `tox.venv.tox_testenv_install_deps` inside of our customized `tox_testenv_install_deps` hook. This keeps things as simple as possible and promotes as much reuse as possible.

But no one asked for my opinion ;).